src/networking/systems.rs

```diff
@@ -0,0 +1,240 @@
+use std::{collections::HashMap};
+
+use bevy::prelude::{Commands, Entity, EventReader, Events, ResMut};
+use quinn::{crypto::rustls::TlsSession, generic::{RecvStream, SendStream}};
+use tokio::{stream::StreamExt, sync::mpsc::UnboundedReceiver, sync::mpsc::UnboundedSender, sync::mpsc::unbounded_channel};
+use tracing::{error, info};
+
+use super::{components::Connection, events::ReceiveEvent, events::SendEvent, id::ConnectionId, id::StreamId, packets::Packet, streams::BoundedPlanetRecvStream, streams::BoundedPlanetSendStream};
+
+/// Internal state of the network session system
+pub struct SessionEventListenerState {
```

Comment:
Able to derive `Debug`, etc?

Reply:
Unfortunately not, `EventReader<SendEvent>` does not implement it.